services/hexathons/src/routes/visit.ts

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable no-underscore-dangle */
+import { apiCall, asyncHandler, BadRequestError, checkAbility } from "@api/common";
+import { Service } from "@api/config";
+import express from "express";
+import { FilterQuery } from "mongoose";
+
+import { VisitModel, Visit } from "../models/visit";
+
+export const visitRouter = express.Router();
+
+// get visit
+visitRouter.route("/").get(
+  checkAbility("read", "Visit"),
+  asyncHandler(async (req, res) => {
+    const company = await apiCall(
+      Service.USERS,
+      { method: "GET", url: `/companies/employees/${req.user?.uid}` },
+      req
+    );
+    const filter: FilterQuery<Visit> = {};
+    if (req.query.hexathon) {
+      filter.hexathon = String(req.query.hexathon);
+    }
+    if (req.query.visitorId) {
+      filter.visitorId = String(req.query.visitorId);
+    }
+    if (company) {
+      filter.company = String(company.id);
+    }
+
+    const visits = await VisitModel.accessibleBy(req.ability).find(filter);
+    return res.status(200).send(visits);
+  })
+);
+
+visitRouter.route("/:visitorId").get(
+  checkAbility("read", "Visit"),
+  asyncHandler(async (req, res) => {
+    // get company of current user
+    const company = await apiCall(
+      Service.USERS,
+      { method: "GET", url: `/companies/employees/${req.user?.uid}` },
+      req
+    );
+
+    if (company) {
+      const { visitorId } = req.params;
+      const visit = await VisitModel.findOne({
+        visitorId,
+        company: company.id,
+      });
+
+      return res.status(200).send(visit?.toJSON());
+    } 
+      throw new BadRequestError("Current user not associated with a company");
+    
+  })
+);
+
+visitRouter.route("/:visitorId").post(
+  checkAbility("create", "Visit"),
+  asyncHandler(async (req, res) => {
+    // get company of current user
+    const company = await apiCall(
+      Service.USERS,
+      { method: "GET", url: `/companies/employees/${req.user?.uid}` },
+      req
+    );
+
+    if (company) {
+      const visit = await VisitModel.create({
+        visitorId: req.params.visitorId,
+        hexathon: req.body.hexathon,
+        company: company.id,
+        employees: req.body.employees,
+        tags: req.body.tags,
+        notes: req.body.notes,
+        time: new Date(),
+      });
+      res.send(visit);
+    } else {
+      throw new BadRequestError("Current user not associated with a company");
+    }
+  })
+);
+
+visitRouter.route("/:visitorId").put(
```

Comment:
I think you can make this update based on `visitId`? So like on the frontend, if you're editing a certain visit, you'll know the object for that id. You can use that instead of `visitorId` and company lookup, and do a direct find query.